src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+/** Deal with taking in user input
+ * and making sense of the input
+ *
+ */
+public class Parser {
+    private TaskList tLst;
+
+    Parser(TaskList tLst) {
+        this.tLst = tLst;
+    }
+
+    /**
+     * Reads user input and determines
+     * what action to carry out
+     * @throws DukeException Thrown if input does not make sense
+     * @throws IOException For interrupted I/O operations
+     */
+    public void read() throws DukeException, IOException {
+        Scanner help = new Scanner(System.in);
+        while (true) {
+            String str = help.nextLine();
+            String first = str.split(" ")[0];
+
+            if (first.equals("bye")) {
+                System.out.println("Bye see you again buddy !");
+                break;
+            } else if (first.equals("list")) {
+                tLst.list();
+            } else if (first.equals("mark")) {
+                Task current = tLst.get(Integer.parseInt(str.split(" ")[1]) - 1);
```

Comment:
I feel like this line may be a bit difficult to read and understand what it does. 
Perhaps you could consider storing the `str.split(" ")[1]) - 1)` part in a separate variable first, since it also appears multiple times in subsequent lines.